### ISSUE TITLE
text invisible regression since nightly 2024-03-19

### ISSUE THREAD

**Issue by zjk261** [ruffle-rs/ruffle#16877]
### Describe the bug

![image](https://github.com/ruffle-rs/ruffle/assets/98996990/873a8863-f9e7-447f-a519-03330b095e7b)

### Expected behavior

text should be visible

### Content Location

https://n0samu.github.io/flash-player-check

### Affected platform

Browser's extension

### Operating system

Windows 10

### Browser

Chrome 126.0.6478.114

**Comment by Lord-McSweeney** [comment]
When I bisect I get that nightly-2023-08-07 was the first commit where the text was invisible.

**Comment by evilpie** [comment]
The text seems to invisible due to this early return:
https://github.com/ruffle-rs/ruffle/blob/dd776d9a8183f52ecdb3775d7e969201c65b0ecc/core/src/display_object/edit_text.rs#L946-L958

`origin.y()` is some positive number and `edit_text.bounds.y_max` is -505.

I think this was original added by @adrian17 in https://github.com/ruffle-rs/ruffle/commit/3c2d629e0d6b8b12b62634cf592ce0f9b48d0c3c. 
/cc @kjarosh

**Comment by kjarosh** [comment]
There was a discussion about it on Discord recently. I believe @britt-j is working on a fix

**Comment by britt-j** [comment]
yeah, I'll be working on getting a test made for this by tomorrow hopefully

**Comment by kjarosh** [comment]
The text is now visible, but text selection is broken. That's the case for every non-left-aligned text, so nothing special in this SWF. This should be fixed after refactoring `screen_position_to_index` so that it uses proper bounds from the layout.
